submissions/iakow/OOP-exercise/index.js

```diff
@@ -0,0 +1,83 @@
+class Inhabitant {
+   constructor({ species, name, say, gender }) {
+      this.species = species;
+      this.name = name;
+      this.say = say;
+      this.gender = gender;
+      this.friends = new Set();
+   }
+
+   makeFriends(...applicants) {
+      applicants.filter(applicant => applicant instanceof Inhabitant && applicant != this)
+         .forEach(applicant => this.friends.add(applicant));
+   }
+
+   toString() {
+      const { name, species, friends, gender, say } = this;
+
+      const friendNames = [];
+      friends.forEach((friend) => friendNames.push(friend.name));
```

Comment:
Here is an equivalent fragment:
```
const friendNames = friends.map(friend => friend.name);
```

Reply:
friends is Set, no Array

Reply:
Makes sense. Then you can leave this as it is, or employ `Array.from`.

In either case let me know if you're gonna fix anything.